shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java

```diff
@@ -0,0 +1,479 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.StandardMethodCodec;
+
+/**
+ * System channel that sends 2-way communication between Flutter and Android to
+ * display Android Views within Flutter.
+ *
+ * Register a {@link PlatformViewsHandler} to implement the Android side of this channel.
+ */
+public class PlatformViewsChannel {
+  private final MethodChannel channel;
+  private PlatformViewsHandler handler;
+
+  private final MethodChannel.MethodCallHandler parsingHandler = new MethodChannel.MethodCallHandler() {
+    @Override
+    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+      // If there is no handler to respond to this message then we don't need to
+      // parse it. Return.
+      if (handler == null) {
+        return;
+      }
+
+      switch (call.method) {
```

Comment:
This method is pretty long which makes it hard for me to follow, lets keep split out the handling of the each different switch case to its own separate methods.

Reply:
done